decryptor/mysql/prepared_statements.go

```diff
@@ -281,3 +285,102 @@ func (m *mysqlBoundValue) Encode() (encoded []byte, err error) {
 
 	return encoded, outErr
 }
+
+// PreparedStatementFieldTracker track and replace DataType for column and param ColumnDefinition
+type PreparedStatementFieldTracker struct {
+	proxyHandler *Handler
+	// shared value that indicates number of param packet
+	paramsCounter int
+	columnsNum    uint16
+}
+
+// NewPreparedStatementFieldTracker create new PreparedStatementFieldTracker
+func NewPreparedStatementFieldTracker(handler *Handler, columnNum uint16) PreparedStatementFieldTracker {
+	return PreparedStatementFieldTracker{
+		proxyHandler: handler,
+		columnsNum:   columnNum,
+	}
+}
+
+// ParamsTrackHandler implements ResponseHandler to track prepare statement params
+func (p *PreparedStatementFieldTracker) ParamsTrackHandler(ctx context.Context, packet *Packet, _, clientConnection net.Conn) error {
+	clientSession := base.ClientSessionFromContext(ctx)
+	if clientSession == nil {
+		p.proxyHandler.logger.Warningln("packet without ClientSession in context")
```

Comment:
```suggestion
		p.proxyHandler.logger.Warningln("Packet without ClientSession in context")
```

all our log entries stars with capital letter